src/convertToObject.js

```diff
@@ -6,7 +6,30 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+
+  const cleanedSource = sourceString.trim().replace(/;\s*$/, '');
+
+  const declarations = cleanedSource
+    .split(';')
+    .filter((declaration) => declaration.trim() !== '');
+
+  declarations.forEach((declaration) => {
+    const colonIndex = declaration.indexOf(':');
+
+    if (colonIndex === -1) {
+      return;
+    }
+
+    const property = declaration.slice(0, colonIndex).trim();
+    const value = declaration.slice(colonIndex + 1).trim();
+
+    if (property && value) {
+      result[property] = value;
+    }
+  });
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. Consider using `map` to transform each declaration into an object, and then `Object.assign` to merge all objects into one.